docroot/modules/custom/foia_webform/src/FoiaSubmissionApiService.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\foia_webform\Plugin\QueueWorker;
+
+use Drupal\foia_webform\AgencyLookupServiceInterface;
+use Drupal\webform\WebformSubmissionInterface;
+use GuzzleHttp\ClientInterface;
+use Drupal\node\Entity\Node;
+
+/**
+ * Class FoiaSubmissionApiService.
+ *
+ * @package Drupal\foia_webform\Plugin\QueueWorker
+ */
+class FoiaSubmissionApiService implements FoiaSubmissionServiceInterface {
+
+  private $apiURL;
+
+  /**
+   * The Agency Lookup service.
+   *
+   * @var \Drupal\foia_webform\AgencyLookupService
+   */
+  protected $agencyLookUpService;
+
+  /**
+   * The HTTP client to make calls to the API with.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * FoiaSubmissionApiService constructor.
+   *
+   * @param \GuzzleHttp\ClientInterface $httpClient
+   *   The HTTP Guzzle client.
+   * @param \Drupal\foia_webform\AgencyLookupServiceInterface $agencyLookupService
+   *   The Agency Lookup service.
+   */
+  public function __construct(ClientInterface $httpClient, AgencyLookupServiceInterface $agencyLookupService) {
+    $this->httpClient = $httpClient;
+    $this->agencyLookUpService = $agencyLookupService;
+  }
+
+  /**
+   * Submits the submission via the API.
+   *
+   * @param \Drupal\webform\WebformSubmissionInterface $webformSubmission
+   *   The webform submission.
+   * @param \Drupal\node\Entity\Node $agencyComponent
+   *   The Agency Components node object.
+   */
+  public function sendSubmissionToComponent(WebformSubmissionInterface $webformSubmission, Node $agencyComponent) {
+    $this->apiURL = $agencyComponent->get('field_submission_api');
+
+    if ($this->apiURL) {
+      $payload = $this->getSubmissionPayload($webformSubmission);
+      $agencyInfo = $this->getAgencyInfo($agencyComponent);
```

Comment:
Is the intent behind the `getSubmissionPayload` method to return the array in the format as defined by the API specification? If so, I would imagine that we'd move the call to `getAgencyInfo` to be within that method so that it can return a payload that's ready to send off.